finetune/model.py

```diff
@@ -216,3 +231,148 @@ def optimizer(lr):
         return tf.estimator.EstimatorSpec(mode=mode, loss=train_loss, eval_metric_ops=metrics)
 
     return _model_fn
+
+
+def get_separate_model_fns(target_model_fn, predict_op, predict_proba_op, build_target_model, build_lm, encoder, target_dim,
```

Comment:
This contains too much duplication with get_model_fn, it will be hard to maintain. Do you think we can refactor so that we have two model functions always, and then a wrapper to merge them together when we only want one?